cmd/terramate/cli/cloud.go

```diff
@@ -220,14 +221,29 @@ func (c *cli) createCloudDeployment(stacks config.List[*config.SortableStack], c
 			logger.Debug().Msg("commit SHA is not being synced because the repository is dirty")
 		}
 
```

Comment:
side note: this func is basically impossible for me to keep in my head and grasp

Reply:
yes, I agree with you. The creation of deployment now has a lot of nuances and just quickly splitting this function would require the functions to receive lots of arguments and return multiple values. Also it has a small optimization to avoid re-parsing/canonicalizing the repository.
It definitely needs polishing. Thanks for pointing it out. I will try to refactor it later on.